utils/billgeneration.js

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable import/imports-first */
+import moment from 'moment';
+import dotenv from 'dotenv-safe';
+
+import {
+  getSolarProjectById,
+  getOwnerById,
+  getRateScheduleById,
+  createSubscriberBill,
+  updateSubscriberBill,
+  getSubscriberBillsByStatementNumber
+} from '../airtable/request';
+import getEnphaseData from './enphase';
+import getLatestBill from './utilityApi';
+
+dotenv.config();
+
+const generateBillForSubscriber = async (subscriber, solarProject) => {
+  // Get data necessary to generate bill
+  console.log(`Generating Bill for ${subscriber.name}`);
+  const {
+    netPgeUsage,
+    ebceRebate,
+    startDate: startMoment,
+    endDate: endMoment
+  } = await getLatestBill(subscriber.meterId);
+
+  const startDate = startMoment.format('MM/DD/YYYY');
+  const endDate = endMoment.format('MM/DD/YYYY');
+
+  let prevBill;
+  if (subscriber.latestBillNumber !== 0) {
+    const bills = await getSubscriberBillsByStatementNumber(
+      subscriber.latestBillNumber
+    );
+    [prevBill] = bills.filter(b => b.subscriberId[0] === subscriber.id);
+  } else {
+    // Edge case for first bill
+    prevBill = {
+      statementNumber: 0,
+      totalEstimatedRebate: 0,
+      balance: 0,
+      startDate: '',
+      endDate: ''
+    };
+  }
+
+  if (
+    startDate === moment(prevBill.startDate).format('MM/DD/YYYY') ||
+    endDate === moment(prevBill.endDate).format('MM/DD/YYYY')
+  ) {
+    console.log('This PG&E bill has already been processed. Skipping!');
+    return;
+  }
+
+  // Set previous bill to previous
+  await updateSubscriberBill(prevBill.id, { status: 'Previous' });
+
+  console.log(
+    `Found PGE Data for ${subscriber.name}
+Net Usage: ${netPgeUsage}
+ebce Rebate: ${ebceRebate}
+start date: ${startDate}
+end date: ${endDate}`
```

Comment:
nit: adjust the indentation for consistency?

Reply:
This weird indentation is actually because with these special strings (and maybe for all strings?) the indentation gets included in the output, so those lines intentionally have no indentation for cleaner logs. Could alternatively format with “\n”s and “+”s but low priority i think